### PR TITLE
Modify builder.go, use third-party pointy to reduce temporary variables

### PR DESCRIPTION
**Pull request by zhengxiexie**
Use pointy to get address of variables, in order to reduce many redundant lines.

### CODE REVIEW

**Review by heypnus — COMMENTED** [review]
A minor comment inline. And I think it's not a "refactor" because the code structure was not changed.

File: pkg/nsx/services/builder.go [hunk 1]
```diff
@@ -44,15 +51,11 @@ func (service *SecurityPolicyService) buildSecurityPolicy(obj *v1alpha1.Security
 	log.V(1).Info("building the model SecurityPolicy from CR SecurityPolicy", "object", *obj)
 	nsxSecurityPolicy := &model.SecurityPolicy{}
 
-	policyID := fmt.Sprintf("sp_%s", obj.UID)
-	nsxSecurityPolicy.Id = &policyID
-
-	policyName := fmt.Sprintf("%s-%s", obj.ObjectMeta.Namespace, obj.ObjectMeta.Name)
-	nsxSecurityPolicy.DisplayName = &policyName
+	nsxSecurityPolicy.Id = AOS(strings.Join([]string{"sp", string(obj.UID)}, "_"))
```
[the remark this answers is not in the document]
Comment (zhengxiexie) [reply pkg/nsx/services/builder.go:54]: `fmt.Sprintf("sp_%s_%d_scope", obj.UID, idx)`, we may be confused with '_' and '-', for consistency, we'd better use only one hyphen.
Surely it's just an option for the suggestion.

**Comment by zhengxiexie** [comment]
> A minor comment inline. And I think it's not a "refactor" because the code structure was not changed.

OK

File: pkg/nsx/services/common/init.go [hunk 2]
```diff
@@ -30,3 +31,8 @@ func init() {
 	Converter = bindings.NewTypeConverter()
 	Converter.SetMode(bindings.REST)
 }
+
+var (
+	AOS = pointy.String // address of string
+	AOI = pointy.Int64  // address of int
```
Comment (heypnus) [inline pkg/nsx/services/common/init.go:37]: How about directly using `String` and `Int64` as the aliases, like what [aws sdk](https://github.com/aws/aws-sdk-go/blob/9d10b7469ebfe21f9ab825461b034f5ac6fc4b8b/aws/convert_types.go#L123) did? We may need not only int64, but also int, int32, etc.. The name like AOI contains too little info about the target type.

File: pkg/nsx/services/builder.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (heypnus) [reply pkg/nsx/services/builder.go]: But for the following cases, I think `strings.Join` introduced the complexity: additional type conversion code, more slices reduced the readability.
`fmt.Sprintf("sp_%s_scope", obj.UID)` -> `strings.Join([]string{"sp", string(obj.UID), strconv.Itoa(idx), "scope"}, "_")`
`fmt.Sprintf("sp_%s_%d_scope", obj.UID, idx)` -> `strings.Join([]string{"sp", string(obj.UID), strconv.Itoa(idx), "scope"}, "_")`
And these changes only write the logic with another writing, which doesn't impact the future code style. If you want to normalize the format, I think you'd better write a normalize function, like generatePolicyID/generatePolicyName, which doesn't accept `"-"` or `"-"` as the parameters, or add a validation function after id/name generation.

File: pkg/nsx/services/common/init.go (hunk 2 above)
Comment (zhengxiexie) [reply pkg/nsx/services/common/init.go:37]: OK

File: pkg/nsx/services/builder.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhengxiexie) [reply pkg/nsx/services/builder.go]: OK, reverted those containing Itoa back.

**Review by heypnus — APPROVED** [review]

**Review by lxiaopei — APPROVED** [review]

**Pull request merged**